Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs

```diff
@@ -61,56 +63,52 @@ private void OnMapData(JSONObject data)
 
             if (Room.ShowingObjects)
                 return;
-
+            
             foreach (var key in data.keys)
             {
                 // player
                 if (key.Length > 2)
-                {
                     SpawnDots(key, data[key].list);
-                }
+
                 else if (key.Equals("k"))
                 {
+                    foreach (var numArray in data[key].list)
+                    {
+                        var x = (int) numArray.list[0].n;
+                        var y = (int) numArray.list[1].n;
+                        _terrainView.addLair(x, y);
+                    }
                     SpawnRoomObjects<SourceKeeperLairView>(data[key].list, SourceKeeperLairView.Path);
                 }
+
                 else if (key.Equals("c"))
-                {
                     SpawnRoomObjects<ControllerView>(data[key].list, ControllerView.Path);
-                }
+
                 else if (key.Equals("s"))
-                {
                     SpawnRoomObjects<SourceView>(data[key].list, SourceView.Path);
-                }
+
                 else if (key.Equals("m"))
-                {
                     SpawnRoomObjects<MineralView>(data[key].list, MineralView.Path);
-                }
+
                 else if (key.Equals("w"))
-                {
                     SpawnRoomObjects<WallView>(data[key].list, WallView.Path);
-                }
+
                 else if (key.Equals("pb"))
-                {
                     SpawnRoomObjects<PowerBankView>(data[key].list, PowerBankView.Path);
-                }
-                else if (key.Equals("p"))
-                {
-                    SpawnRoomObjects<PortalView>(data[key].list, PortalView.Path);
-                }
             }
         }
 
-        private void SpawnRoomObjects<T>(List<JSONObject> list, string prefabPath)
-            where T : IMapViewComponent
+        private void SpawnRoomObjects<T>(List<JSONObject> list, string PrefabPath)
```

Comment:
Should not be capital P in a variable name :P

Reply:
noted